src/renderer/api/kube-object.ts

```diff
@@ -78,15 +79,59 @@ export class KubeObject implements ItemObject {
     return !item.metadata.name.startsWith("system:");
   }
 
-  static isJsonApiData(object: any): object is KubeJsonApiData {
-    return !object.items && object.metadata;
-  }
-
-  static isJsonApiDataList(object: any): object is KubeJsonApiDataList {
-    return object.items && object.metadata;
-  }
-
-  static stringifyLabels(labels: { [name: string]: string }): string[] {
+  static isJsonApiData(object: unknown): object is KubeJsonApiData {
+    return (
+      isObject(object)
+      && hasTypedProperty(object, "kind", isString)
```

Comment:
Why do we need to test type of these properties? Don't we trust that kubernetes will respond with correct types? 🤔

Reply:
I would consider this a form of defence in depth. With this here we are making sure that we are not messing with the data too much with any of the other code. 

Yes I think that kubernetes will probably respond with the correct data fields.

Reply:
To me it feels like a bit too much defence... it's like validating data from a database every time it's read. Not sure what others think? /cc @nevalla @jim-docker @ixrock  

If we really want to validate data we probably should use something like https://joi.dev/ .

Reply:
Why would you not want to validate data from the db? In most languages (except dynamic typed ones) that is basically a necessity. I think that validating data as soon as possible is a great way to prevent bad data from getting through.

Reply:
Starts to to a bit off-topic but if a database has a column with a type it cannot really be anything else.. so why would you validate the type on read?

Reply:
To make sure the program requesting the data has the same model as the database?

Plus that only works for strict typed databases. Which an HTTP request is not that sort imo.

Reply:
>  In most languages (except dynamic typed ones) that is basically a necessity.

Maybe then we should re-write Lens on c++? :D

I agree with @jakolehm that it's a bit too much defence..

Reply:
Is it going to make life more difficult for programmers/users? Is it easy to draw the line at less defence? If so then we should back it off. Otherwise it's pretty thorough but I didn't think it was too much...

Reply:
I would also like to make the counter point that just because we can do less validation doesn't mean that we should. For instance if something like `selfLink` becoming optional and deprecated happens again then users (or CI tests if we get them that good) will start seeing errors a lot sooner.

Reply:
@jakolehm Would you prefer if I reworked this PR in terms of `joi` then?

Reply:
I'm not completely happy that we are checking data types here and doing it with our custom implementation but the implementation is hidden behind functions so we can improve/cleanup later.